### PR TITLE
[13.0] [MIG] Account Fiscal Position Rule Purchase

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Based on https://github.com/OCA/account-fiscal-rule/pull/81

### CODE REVIEW

**Comment by dreispt** [comment]
Any chance you could split the black, etc and the actual migration changes into two separate commits?

**Comment by ThomasBinsfeld** [comment]
> Any chance you could split the black, etc and the actual migration changes into two separate commits?

Of course, done.

File: account_fiscal_position_rule_purchase/models/purchase.py [hunk 1]
```diff
@@ -17,16 +17,18 @@ def _fiscal_position_map(self, **kwargs):
...
-
-        kwargs = {
+    def _prepare_fiscal_position_map_kwargs(self):
+        self.ensure_one()
+        return {
             "company_id": self.company_id,
             "partner_id": self.partner_id,
             "partner_invoice_id": self.partner_id,
-            "partner_shipping_id": self.partner_id,
+            "partner_shipping_id": self.dest_address_id,
```
Comment (dreispt) [inline account_fiscal_position_rule_purchase/models/purchase.py:26]: Does this make sense?
```suggestion
            "partner_shipping_id": self.dest_address_id or self.partner_id,
```

**Review by dreispt — APPROVED** [review]
LGTM, one suggestion to consider

**Review by chrisdec — APPROVED** [review]
Functional test

**Pull request merged**